File: BloodFloater/src/BloodFloater.Services.Impl/DonorService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using BloodFloater.DAL;
using BloodFloater.Models;

namespace BloodFloater.Services.Impl
{
    public class DonorService : BaseService<Donor, ViewModels.Donor>, IDonorService
    {
        public DonorService(IDonorRepository repository) : base(repository)
        {
        }
    }
}

File: BloodFloater/src/BloodFloater.Web.UI/wwwroot/src/pages/donor/donor.service.ts
import {IBaseService, BaseService} from '../../app/services/base.service'
import {Injectable, Optional} from '@angular/core';
import {Donor} from '../../app/models';
import { Http, URLSearchParams } from '@angular/http';

export interface IDonorService extends IBaseService<Donor> {
    getByProfileId(id : number);
}

@Injectable()
export class DonorService extends BaseService <Donor> implements IDonorService {
    constructor(public http : Http) {
        super(http, 'donors');
    }

    getByProfileId(id: number){
        var params = new URLSearchParams();
        params.set('profileId', id.toString());
        return this.getByQuery(params);
    }
}